nwb_conversion_tools/datainterfaces/behavior/movie/movie_utils.py

```diff
@@ -1,21 +1,146 @@
-"""Authors: Cody Baker."""
+"""Authors: Saksham Sharda, Cody Baker."""
 from pathlib import Path
+from typing import Union, Tuple
+
 import numpy as np
-from typing import Union
+from tqdm import tqdm
+
+from ....utils.json_schema import FilePathType
 
 try:
     import cv2
 
     HAVE_OPENCV = True
 except ImportError:
     HAVE_OPENCV = False
+INSTALL_MESSAGE = "Please install opencv to use this interface! (pip install opencv-python)"
 
 PathType = Union[str, Path]
 
 
+class VideoCaptureContext:
+    """Retrieving video metadata and frames using a context manager"""
+
+    def __init__(self, file_path: FilePathType):
+        assert HAVE_OPENCV, INSTALL_MESSAGE
+        self.vc = cv2.VideoCapture(filename=file_path)
+        self.file_path = file_path
+        self._current_frame = 0
+        self._frame_count = None
```

Comment:
```suggestion
        self._frame_count = self._movie_frame_count()
```
Would this make sense to do?

Reply:
I wanted to keep the `__init__` as light as possible, not retrieving anything from the movie other than instantiating the `VideoWriter` object.